src/main/scala/rocket/Frontend.scala

```diff
@@ -318,55 +325,55 @@ class FrontendModule(outer: Frontend) extends LazyModuleImp(outer)
     when (useRAS) {
       predicted_npc := btb.io.ras_head.bits
     }
-    when (fq.io.enq.fire() && (s2_btb_taken || taken)) {
-      s2_partial_insn_valid := false
+    when (fq.io.enq.fire && (s2_btb_taken || taken)) {
+      s2_partial_insn_valid := false.B
     }
     when (!s2_btb_taken) {
       when (taken) {
         fq.io.enq.bits.btb.bridx := taken_idx
-        fq.io.enq.bits.btb.taken := true
-        fq.io.enq.bits.btb.entry := UInt(tileParams.btb.get.nEntries)
-        when (fq.io.enq.fire()) { s2_redirect := true }
+        fq.io.enq.bits.btb.taken := true.B
+        fq.io.enq.bits.btb.entry := tileParams.btb.get.nEntries.U
+        when (fq.io.enq.fire) { s2_redirect := true.B }
       }
     }
 
     assert(!s2_partial_insn_valid || fq.io.enq.bits.mask(0))
-    when (s2_redirect) { s2_partial_insn_valid := false }
-    when (io.cpu.req.valid) { wrong_path := false }
+    when (s2_redirect) { s2_partial_insn_valid := false.B }
+    when (io.cpu.req.valid) { wrong_path := false.B }
   }
 
   io.cpu.resp <> fq.io.deq
 
   // supply guest physical address to commit stage
   val gpa_valid = Reg(Bool())
   val gpa = Reg(UInt(vaddrBitsExtended.W))
-  when (fq.io.enq.fire() && s2_tlb_resp.gf.inst) {
+  when (fq.io.enq.fire && s2_tlb_resp.gf.inst) {
     when (!gpa_valid) {
       gpa := s2_tlb_resp.gpa
     }
-    gpa_valid := true
+    gpa_valid := true.B
   }
   when (io.cpu.req.valid) {
-    gpa_valid := false
+    gpa_valid := false.B
   }
   io.cpu.gpa.valid := gpa_valid
   io.cpu.gpa.bits := gpa
 
   // performance events
-  io.cpu.perf := icache.io.perf
```

Comment:
why

Reply:
perf on the CPU side have an additional field `tlbMiss` (See below)
This is essentially a partial assignment.